gr-blocks/lib/pack_k_bits_bb_impl.cc

```diff
@@ -40,8 +42,22 @@ int pack_k_bits_bb_impl::work(int noutput_items,
     const unsigned char* in = (const unsigned char*)input_items[0];
     unsigned char* out = (unsigned char*)output_items[0];
 
+    std::vector<tag_t> wintags; // Temp variable to store tags for prop
+
+    // GR_LOG_DEBUG(d_logger, boost::format("Packing Outputs"));
     d_pack.pack(out, in, noutput_items);
 
+    // Propagate tags
+    get_tags_in_range(wintags, 0, nitems_read(0), nitems_read(0) + (noutput_items * d_k));
+
+    // GR_LOG_DEBUG(d_logger, boost::format("Propagating tags"));
+    std::vector<tag_t>::iterator t;
+    for (t = wintags.begin(); t != wintags.end(); t++) {
+        tag_t new_tag = *t;
```

Comment:
No need to copy the tags – `get_tags_in_range` fills your vector with tags you own!

Reply:
Are you saying a better solution would be to just edit the offset of `t` directly and call `add_item_tag(0,t)`?

I largely copied this from the block execution code (ie, where the 'automatic' tag propagation happens).  My thought was the 'copy' to `new_tag` was to allow editing the offset.  I assumed it had something to do with directly editing the target object of the iterator but I'm also not really sure...

Reply:
jep, I'd say that would be the more performant method. HOWEVER, I did accept this because this is a common pattern in the codebase and ecosystem, and it's really not that expensive, and honestly, this patch is good and should be merged. We can improve the perf separately.